metals/src/main/scala/scala/meta/internal/metals/codeactions/RenameImport.scala

```diff
@@ -0,0 +1,92 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.Importee
+import scala.meta.Name
+import scala.meta.Tree
+import scala.meta.inputs.Position
+import scala.meta.internal.metals.CodeAction
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.parsing.Trees
+import scala.meta.pc.CancelToken
+
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.{lsp4j => l}
+import scala.meta.internal.metals.ClientCommands
+
+class RenameImport(trees: Trees) extends CodeAction {
+
+  import RenameImport._
+  override def kind: String = l.CodeActionKind.Refactor
+
+  override def contribute(
+      params: l.CodeActionParams,
+      token: CancelToken
+  )(implicit ec: ExecutionContext): Future[Seq[l.CodeAction]] = Future {
+
+    def findLastEnclosingAtPos(tree: Tree, pos: Position): Option[Name] = {
+      tree.children.find { child =>
+        child.pos.start <= pos.start && pos.start <= child.pos.end
+      } match {
+        case None =>
+          tree match {
+            case name: Name => Some(name)
+            case _ => None
+          }
+        case Some(value) => findLastEnclosingAtPos(value, pos)
+      }
+    }
+
+    def renameImportCodeAction(name: Name): Option[l.CodeAction] =
+      name.parent.flatMap {
+        case nm @ Importee.Name(_) =>
+          val range = nm.pos.toLSP
+          val text = s"{ $name => }"
```

Comment:
So I'm not fully sure how this works with cursor position. For example I tried it in both nvim-metals and coc-metals just to see how two different clients would handle the codeAction edit and the cursor seems to stay in place for both of them.


### nvim-metals

![2021-03-04 20 09 50](https://user-images.githubusercontent.com/13974112/110017364-6adc8c00-7d26-11eb-8880-7c7a5e102065.gif)


### coc-metals

![2021-03-04 20 14 37](https://user-images.githubusercontent.com/13974112/110017379-716b0380-7d26-11eb-947e-12e483fbf012.gif)